src/main/drivers/pwm_output_dshot.c

```diff
@@ -141,17 +262,34 @@ static void motor_DMA_IRQHandler(dmaChannelDescriptor_t *descriptor)
         }
 
         DMA_CLEAR_FLAG(descriptor, DMA_IT_TCIF);
+#ifdef USE_DSHOT_TELEMETRY
+        if (dshotTelemetryEnabled) {
+            pwmDshotSetDirectionOutput(motor, false, NULL, NULL);
+
+            DMA_SetCurrDataCounter(motor->timerHardware->dmaRef, motor->useProshot ? PROSHOT_TELEMETRY_INPUT_LEN : DSHOT_TELEMETRY_INPUT_LEN);
+            DMA_Cmd(motor->timerHardware->dmaRef, ENABLE);
+            TIM_DMACmd(motor->timerHardware->tim, motor->timerDmaSource, ENABLE);
+        }
+#endif
     }
 }
 
-void pwmDshotMotorHardwareConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, motorPwmProtocolTypes_e pwmProtocolType, uint8_t output)
+FAST_CODE static void pwmDshotSetDirectionOutput(motorDmaOutput_t * const motor, bool output, TIM_OCInitTypeDef *pOcInit, DMA_InitTypeDef* pDmaInit)
```

Comment:
That does sound more like natural language, doesn't it :-). I'll address it in the dshot_telemetry PR if that needs another update. Or otherwise later on here.